share/doc/wake/from-scala.adoc

```diff
@@ -0,0 +1,396 @@
+= Learning Wake as a Scala programmer
+:toc:
+
+This document describes Wake from the perspective of a programmer who is already familiar with Scala.
+Wake is a functional programming language, and it is very similar to the functional aspects of Scala.
+
+Wake is, however, "purer" than Scala,
+so while Scala allows for a blend of functional, imperative, and object-oriented styles,
+Wake more strictly enforces a functional style.
+In fact, the only side effects in Wake are printing, running subprocesses, and panicking.
+
+Since Wake does not support function overloading, the same function name cannot be defined more than once with different arguments.
+Therefore many of the built-in collection types define functions that are prefixed with the first initial of the type name.
+For example, while the `List` type has a `map` function, the `Option` type has an `omap` function, and the `Result` type has an `rmap` function.
+
+See the following sections to learn about how various concepts in Scala map to Wake.
+
+
+== Primitive types
+
+Wake generally has familiar syntax for primitive types.
+Syntax for more advanced functionality of strings -- such as interpolation and multi-line strings -- differs between Wake and Scala.
+See <<table-primitives>> for more information.
+
+[[table-primitives]]
+.Primitive types in Scala and Wake
+[%autowidth.stretch,cols="m,a,m,a"]
+|===
+| Scala Type | Scala Syntax | Wake Type | Wake Syntax
+
+
+| Int
+|
+[source,scala]
+----
+1
+----
+| Integer
+|
+[source,wake]
+----
+1
+----
+
+
+| Double
+|
+[source,scala]
+----
+1.0
+----
+| Double
+|
+[source,wake]
+----
+1.0
+----
+
+
+| String
+|
+[source,scala]
+----
+"Plain string"
+
+s"${Interpolated} string"
+
+"""
+Multi-
+line
+string
+"""
+----
```

Comment:
Scala has multiline strings with interpolation:

```scala
s"""
Multi-line string
with ${interpolation}
"""
```

It might be worth talking about Scala strip prefix vs. Wake's multi-line string start, eg.:
```scala
    val x = s"""
      |Each line
      |starts at the pipe
      |""".stripPrefix
```
vs.
```
    def x = "%
      Each line
      starts lined up with the first line
      %"
```
I might have Wake's syntax slightly wrong, but it's relevant, possibly worth just putting in a topic-document though rather than here.

Reply:
Thanks, I added one more example with the Scala multi-line string with interpolation, but I think the remainder of the string features belong in a document specific to string handling in Wake.